src/WebJobs.Script.WebHost/Management/InstanceManager.cs

```diff
@@ -150,7 +150,7 @@ public bool StartAssignment(HostAssignmentContext context)
                 _webHostEnvironment.DelayRequests();
 
                 // start the specialization process in the background
-                Task.Run(async () => await Assign(context));
+                Task.Run(async () => await Assign(context)).Wait();
```

Comment:
@fabiocav while the specialization is running in the background, the endpoint /pod/assign is returning a 202 due to which a http request is received by the host before specialization is complete. The change here helped me confirm the same. Any thoughts on this?

Reply:
@fabiocav  @pragnagopa any thoughts on this?

Reply:
I can close this PR. This is no loner relevant. We fixed it using https://github.com/Azure/azure-functions-host/pull/8493